moduleframework/mtf_environment.py

```diff
@@ -29,7 +29,12 @@
 from moduleframework.environment_prepare.rpm_prepare import EnvRpm
 from moduleframework.environment_prepare.nspawn_prepare import EnvNspawn
 from moduleframework.environment_prepare.openshift_prepare import EnvOpenShift
+from moduleframework import mtf_scheduler
 
+# I'm lazy to do own argument parser here.
+args, unknown = mtf_scheduler.cli()
```

Comment:
why unknown? It will show this message in case wrong argument.
~~~
$ mtf --skipt
MODULE=None ; we support ['rpm', 'nspawn', 'docker', 'openshift'] 
 === expecting your magic, enjoy! === 
Failed to load plugin from module "avocado_runner_remote": ImportError('cannot import name TestName',)
Failed to load plugin from module "avocado_runner_docker": ImportError('cannot import name TestName',)
usage: avocado run [-h] [-d] [--force-job-id UNIQUE_JOB_ID]
                   [--job-results-dir DIRECTORY] [--job-timeout SECONDS]
                   [--failfast {on,off}] [--keep-tmp {on,off}]
                   [--ignore-missing-references {on,off}] [--sysinfo {on,off}]
                   [--execution-order {tests-per-variant,variants-per-test}]
                   [-s] [--show-job-log]
                   [--store-logging-stream [STREAM[:LEVEL] [STREAM[:LEVEL]
                   ...]]] [--log-test-data-directories]
                   [--output-check-record {none,stdout,stderr,both,combined,all}]
                   [--output-check {on,off}]
                   [--loaders [LOADERS [LOADERS ...]]]
                   [--external-runner EXECUTABLE]
                   [--external-runner-chdir {runner,test}]
                   [--external-runner-testdir DIRECTORY] [-t TAGS]
                   [--filter-by-tags-include-empty] [--env-keep ENV_KEEP]
                   [--gdb-run-bin EXECUTABLE[:BREAKPOINT]]
                   [--gdb-prerun-commands EXECUTABLE:COMMANDS]
                   [--gdb-coredump {on,off}] [--html FILE] [--open-browser]
                   [--html-job-result {on,off}] [--journal] [--json FILE]
                   [--json-job-result {on,off}] [--replay REPLAY_JOBID]
                   [--replay-test-status REPLAY_TESTSTATUS]
                   [--replay-ignore REPLAY_IGNORE] [--replay-resume]
                   [--tap FILE] [--tap-job-result {on,off}]
                   [--wrapper SCRIPT[:EXECUTABLE]] [--xunit FILE]
                   [--xunit-job-result {on,off}] [-z]
                   [TEST_REFERENCE [TEST_REFERENCE ...]]

positional arguments:
  TEST_REFERENCE        List of test references (aliases or paths)

optional arguments:
  -h, --help            show this help message and exit
  -d, --dry-run         Instead of running the test only list them and log
                        their params.

~~~

I would prefer to fix cli instead of ValueError exception.

Reply:
This is okay. I've didn't modify this command in case of ``mtf-env-`` commands it leads to your solution.
In case of ``mtf`` all unknown args are passed to avocado thats why this message.
But as I said this code is not in mtf-scheduler but in mtf-environmnet,